### PR TITLE
[PROD-554] Upgrade Python & project dependencies

### PR DESCRIPTION
**Pull request by rmoneys**
https://synccomputing.atlassian.net/browse/PROD-554

To create a new python environment:
```
% pyenv install 3.11.1
% pyenv local 3.11.1
% python --version
Python 3.11.1
% python -m venv ~/.virtualenvs/spark_log_parser311
% . ~/.virtualenvs/spark_log_parser311/bin/activate
% pip install '.[dev]'
```

### CODE REVIEW

File: spark_log_parser/parsing_models/application_model_v2.py [hunk 1]
```diff
@@ -105,7 +105,7 @@ def save_to_s3(self, saveDat, filepath, compress):
 DataType = TypeVar("DataType")
 
 
-class AbstractSparkApplicationDataLoader(abc.ABC, Generic[DataType], DataLoader):
+class AbstractSparkApplicationDataLoader(DataLoader, Generic[DataType], abc.ABC):
```
Comment (rmoneys) [inline spark_log_parser/parsing_models/application_model_v2.py:108]: In Python 3.11.1 this yields,
```
TypeError: Cannot create a consistent method resolution order (MRO) for bases object, Generic, DataLoader
```
It's typically better to put the less general types first.

File: pyproject.toml [hunk 2]
```diff
@@ -17,29 +17,30 @@ classifiers = [
...
+    "pydantic==1.10.4",
+    "requests==2.28.1",
+    "aiodataloader==0.3.0"
 ]
 dynamic = ["version", "description"]
 
 [project.optional-dependencies]
-test = [
-    "pytest==7.1.2",
+dev = [
```
Comment (gseyffert-sync) [inline pyproject.toml:35]: :+1:

File: spark_log_parser/parsing_models/application_model_v2.py (hunk 1 above)
Comment (gseyffert-sync) [reply spark_log_parser/parsing_models/application_model_v2.py:108]: Ah, I was going to ask! Makes sense to me :+1:

**Review by gseyffert-sync — APPROVED** [review]

**Review by gseyffert-sync — APPROVED** [review]

**Comment by gseyffert-sync** [comment]
Thanks for doing this!

**Pull request merged**